tabular/src/autogluon/tabular/models/vowpalwabbit/vowpalwabbit_model.py

```diff
@@ -0,0 +1,288 @@
+        if loss_function:
+            if self.problem_type in PROBLEM_TYPES_CLASSIFICATION:
+                assert loss_function in self.CLASSIFICATION_LOSS_FUNCTIONS, \
+                    f'For {self.problem_type} problem, VW supports: {self.CLASSIFICATION_LOSS_FUNCTIONS}. ' \
+                    f'Got loss_function:{loss_function}'
+            elif self.problem_type in PROBLEM_TYPES_REGRESSION:
+                assert loss_function in self.REGRESSION_LOSS_FUNCTIONS, \
+                    f'For {self.problem_type} problem, VW supports: {self.REGRESSION_LOSS_FUNCTIONS}. ' \
+                    f'Got loss_function:{loss_function}'
+
+    def _get_default_loss_function(self):
+        # Ref: https://github.com/VowpalWabbit/vowpal_wabbit/wiki/Loss-functions
+        if self.problem_type in PROBLEM_TYPES_CLASSIFICATION:
+            return 'logistic'
+        elif self.problem_type in PROBLEM_TYPES_REGRESSION:
+            return 'squared'
+
+    def save(self, path: str = None, verbose=True) -> str:
+        """
+        AutoGluon by default saves the complete Abstract Model in a pickle file format.
+        This includes the internal self.model which is the actual model.
+        However, saving VW model in pickle is not possible.
+        Hence, we dump the Abstract Model by setting setting self.model as None
+        and save self.model as a separate internal file using that model's saving mechanism
+
+        :param path: path where model is to be saved
+        :param verbose: verbosity
+        :return: path where model is saved
+        """
+
+        self._load_model = self.model is not None
+        __model = self.model
+        self.model = None
+        path = super().save(path=path, verbose=verbose)
+        self.model = __model
+        # Export model
+        if self._load_model:
+            file_path = path + self.model_internals_file_name
+            self.model.save(file_path)
+        self._load_model = None
+        return path
+
+    @classmethod
+    def load(cls, path: str, reset_paths=True, verbose=True):
+        """
+        There are two files which needs to be loaded.
+        First is the Abstract Model pickle dump and second is the internal model file.
+        For VW, based on different problem_type/hyperparams, loading arguments will be different
+        """
+        try_import_vowpalwabbit()
+        from vowpalwabbit import pyvw
+        # Load Abstract Model. This is without the internal model
+        model = super().load(path, reset_paths=reset_paths, verbose=verbose)
+        params = model._get_model_params()
+        # Load the internal model file
+        if model._load_model:
+            file_path = path + cls.model_internals_file_name
+
+            model_load_params = f" -i {file_path}"
+            if model.problem_type in PROBLEM_TYPES_CLASSIFICATION:
+                model_load_params += " --probabilities"
+            if params[cls.sparse_weights]:
```

Comment:
Why not have a default value? Default can align with the default in VowpalWabbit

```
if params.get(cls.sparse_weights, DEFAULT):
```

Reply:
sparse_weights param is being set with default params, so this will never have a None value or key being missed: https://github.com/killerSwitch/autogluon/blob/vw-algorithm/tabular/src/autogluon/tabular/models/vowpalwabbit/vowpalwabbit_model.py#L260